controllers/status_controller.go

```diff
@@ -107,16 +109,61 @@ func (r *StatusReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, fmt.Errorf("failed to update lbSocket of Postgres: %w", err)
 	}
 
+	// Fetch the list of operator-generated secrets
+	secrets := &corev1.SecretList{}
+	if err := r.List(ctx, secrets, owner.ToUserPasswordsSecretListOption()...); client.IgnoreNotFound(err) != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to fetch the list of operator generated secrets: %w", err)
+	}
+
+	if err := r.createOrUpdateSecret(ctx, &owner, secrets, log); err != nil {
+		return ctrl.Result{}, err
+	}
+
 	return ctrl.Result{}, nil
 }
 
 // SetupWithManager registers this controller for reconciliation of Postgresql resources
 func (r *StatusReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&zalando.Postgresql{}).
+		// Owns(&corev1.Secret{}). // todo: better the watching on secrets that weren't generated by this controller
 		Complete(r)
 }
 
+func (r *StatusReconciler) createOrUpdateSecret(ctx context.Context, in *pg.Postgres, secrets *corev1.SecretList, log logr.Logger) error {
+	secret, err := in.ToUserPasswordsSecret(secrets, r.Scheme)
+	if err != nil {
+		return fmt.Errorf("failed to convert to the secret containing user password pairs: %w", err)
+	}
+
+	// placeholder of the secret with the specified namespaced name
+	fetched := &corev1.Secret{ObjectMeta: secret.ObjectMeta}
+	// todo: update to CreateOrPatch()
+	result, err := controllerutil.CreateOrUpdate(ctx, r.Control, fetched, func() error {
+		fetched.Data = secret.Data
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create or update the secret containing user password pairs: %w", err)
+	}
+	log.Info("secret created or updated", "operation result", result)
+
+	if result == controllerutil.OperationResultCreated {
```

Comment:
You're right here as well: The name will probably not change and could be derived automativally (by convention). Hence, we would not need the field secretName in the spec.

What we should do, however, is export the String-Suffix we use to generate the secret name (currently `-password`) so it the same string will be used in the `cloudctl`/cloud-api

Reply:
@majst01
We added an exported func `ToUserPasswordsSecretName` to `Postgres`. Can we remove `secretName` from the spec.?